### PR TITLE
feat: replace `ln` with an environment variable to set the /photos folder (main)

### PR DESCRIPTION
**Pull request by martabal**
Replace the symbolic link which links the two `./upload` folders with /photos by the environment variable `IMMICH_MEDIA_LOCATION` [introduced in the last version](https://github.com/immich-app/immich/pull/2047).

### CODE REVIEW

**Comment by hydazz** [comment]
This is a breaking change? migration steps?

**Comment by martabal** [comment]
Indeed this is a breaking change, we will keep symlinks for a moment. To update file paths in database, you need to run the following jobs : 
- storage migration 
- generate thumbnails 
- transcode videos

Generating thumbnails and transcoding videos are resource-intensive tasks with processing time depending on the size of the library.

https://github.com/immich-app/immich/blob/34d300d1da296721d647c751169c4441ba33ec7d/server/libs/domain/src/storage-template/storage-template.core.ts#L55-L68

Immich has a mechanism to check the differences between the source and destination.

**Comment by hydazz** [comment]
@martabal

this code seems to reliably update the db:
```python
# pip3 install psycopg2-binary
import psycopg2

# Define connection parameters
host = '192.168.1.2'
port = '5432'
username = 'postgres'
password = 'postgres'
database = 'immich'

# Connect to the database
conn = psycopg2.connect(
    host=host,
    port=port,
    user=username,
    password=password,
    dbname=database
)

# Open a cursor to perform database operations
cur = conn.cursor()

# Define a SQL query to update the rows in the assets table
sql = "UPDATE assets SET \"originalPath\" = REPLACE(\"originalPath\", 'upload/', '/photos/'), \"resizePath\" = REPLACE(\"resizePath\", 'upload/', '/photos/'), \"webpPath\" = REPLACE(\"webpPath\", 'upload/', '/photos/'), \"encodedVideoPath\" = REPLACE(\"encodedVideoPath\", 'upload/', '/photos/') WHERE \"originalPath\" LIKE 'upload/%';"

# Execute the SQL query
cur.execute(sql)

# Commit the changes to the database
conn.commit()

# Close the cursor and database connection
cur.close()
conn.close()
```

**Comment by martabal** [comment]
For how long do we keep the migration script ?

**Comment by hydazz** [comment]
Probably like a month

File: root/etc/s6-overlay/s6-rc.d/init-config-immich/run [hunk 1]
```diff
@@ -7,7 +7,7 @@ mkdir -p \
     /run/immich
 
 # update the database to reference the correct paths, now that 'IMMICH_MEDIA_LOCATION' is set
-if [[ -z "${TEST_RUN}" ]]; then
+if [[ -z "${DISABLE_IG_MIGRATION}" ]]; then
```
Comment (hydazz) [inline root/etc/s6-overlay/s6-rc.d/init-config-immich/run:10]: this variable was just so ci works

Comment (martabal) [reply root/etc/s6-overlay/s6-rc.d/init-config-immich/run:10]: Ok I revert it

**Pull request merged**